src/core/qgstessellator.cpp

```diff
@@ -412,9 +481,13 @@ void QgsTessellator::addPolygon( const QgsPolygon &polygon, float extrusionHeigh
     const QgsPoint ptStart( exterior->startPoint() );
     const QgsPoint pt0( QgsWkbTypes::PointZ, ptStart.x(), ptStart.y(), std::isnan( ptStart.z() ) ? 0 : ptStart.z() );
 
+    const float scaleX = !mBounds.isNull() ? 10000.0 / mBounds.width() : 1.0;
+    const float scaleY = !mBounds.isNull() ? 10000.0 / mBounds.height() : 1.0;
```

Comment:
Is this arbitrarily chosen value to scale the coordinates to 0...10000 ?

Reply:
Yes.. it's a magic number based on.... Nothing in particular. Maybe it should even be higher?

Reply:
I was actually thinking originally to scale it to lower numbers... e.g. 0...1 but then realized it will likely not improve the numerical stability.

Reply:
That was my V1. But it still encountered the precision warnings and no results. I guess I'll bump it further if we hit issues...